core/src/main/java/org/apache/carbondata/core/datastore/filesystem/S3CarbonFile.java

```diff
@@ -60,73 +58,57 @@ object and then performs rename(copy).
           Refer CARBONDATA-2670 for tracking this.
    */
   @Override
-  public boolean renameForce(String changeToName) {
-    FileSystem fs;
+  public boolean renameForce(String changetoName) {
     try {
-      deleteFile(changeToName, FileFactory.getFileType(changeToName));
-      fs = fileStatus.getPath().getFileSystem(hadoopConf);
-      return fs.rename(fileStatus.getPath(), new Path(changeToName));
+      deleteFile();
```

Comment:
This seems wrong, it is deleting the current file before renaming it

Reply:
changed